libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs

```diff
@@ -86,8 +87,13 @@ public virtual async Task<string> SendMessageAsync(string path, FacebookMessage
                         var stringResponse = JsonConvert.DeserializeObject<FacebookResponseOk>(responseBody);
                         return stringResponse.MessageId;
                     }
+                    else
+                    {
+                        // Return error info
+                        return res.ToString();
+                    }
```

Comment:
This 'else' is not needed. 

Also, it seems like the method's intent is to return either a message Id, if the request is successful, or an empty string, if not. This introduces a change of behavior. Is that what you mean?

Reply:
Sorry, that was a way to debug. I have now changed this to throw an exception so the error is visible in Application Insights